e2e/rescue/claim.spec.ts

```diff
@@ -0,0 +1,118 @@
+import type { Page } from "@playwright/test";
+import fs from "fs";
+
+import dict from "../../src/i18n/i18n";
+import {
+    elementsSendToAddress,
+    generateBitcoinBlock,
+    generateLiquidBlock,
+    getBitcoinAddress,
+    waitForNodesToSync,
+} from "../utils";
+
+const getCurrentSwapId = (page: Page) => {
+    const url = new URL(page.url());
+    return url.pathname.split("/").pop();
+};
+
+const createChainSwap = async (page: Page) => {
+    await page.goto("/");
+
+    const assetSelector = page.locator("div[class='asset asset-LN'] div");
+    await assetSelector.click();
+
+    const lbtcAsset = page.locator("div[data-testid='select-L-BTC']");
+    await lbtcAsset.click();
+
+    const receiveAmount = "0.01";
+    const inputReceiveAmount = page.locator(
+        "input[data-testid='receiveAmount']",
+    );
+    await inputReceiveAmount.fill(receiveAmount);
+
+    const inputSendAmount = page.locator("input[data-testid='sendAmount']");
+    const sendAmount = "0.01003057";
+    await expect(inputSendAmount).toHaveValue(sendAmount);
+
+    const inputOnchainAddress = page.locator(
+        "input[data-testid='onchainAddress']",
+    );
+    await inputOnchainAddress.fill(await getBitcoinAddress());
+
+    const buttonCreateSwap = page.locator(
+        "button[data-testid='create-swap-button']",
+    );
+    await buttonCreateSwap.click();
+
+    return sendAmount;
+};
+
+test.describe("Claim", () => {
+    test.beforeEach(async () => {
+        await generateLiquidBlock();
+    });
+
+    test("Claim pending chain swap via rescue key scan", async ({ page }) => {
+        const sendAmount = await createChainSwap(page);
+
+        const downloadPromise = page.waitForEvent("download");
+        await page
+            .getByRole("button", { name: dict.en.download_new_key })
+            .click();
+
+        const fileName = "rescue-file.json";
+        await (await downloadPromise).saveAs(fileName);
+
+        await page.getByTestId("rescueFileUpload").setInputFiles(fileName);
+
+        await page.locator("p[data-testid='copy-box']").click();
+
+        const address = await page.evaluate(() => {
+            return navigator.clipboard.readText();
+        });
+
+        const swapId = getCurrentSwapId(page);
+        // Clear storage to generate new rescue key
+        await page.evaluate(() => window.localStorage.clear());
+        await page.reload();
+
+        await elementsSendToAddress(address, sendAmount);
+
+        await generateLiquidBlock();
+        await waitForNodesToSync();
+
+        await generateBitcoinBlock();
+        await waitForNodesToSync();
+
+        await page.getByRole("link", { name: "Rescue" }).click();
+
+        await page
+            .getByRole("button", { name: "Rescue external swap" })
+            .click();
+
+        await page.getByTestId("refundUpload").setInputFiles(fileName);
+
+        if (fs.existsSync(fileName)) {
+            fs.unlinkSync(fileName);
+        }
```

Comment:
This should run after each test, not in here
https://playwright.dev/docs/api/class-test#test-after-each